pgpubsub/channel.py

```diff
@@ -147,50 +148,61 @@ class TriggerChannel(BaseChannel):
     @classmethod
     def deserialize(cls, payload: Union[Dict, str]):
         payload_dict = super().deserialize(payload)
-        old_model_data = cls._build_model_serializer_data(payload_dict, state='old')
-        new_model_data = cls._build_model_serializer_data(payload_dict, state='new')
-
-        old_deserialized_objects = serializers.deserialize(
-            'json',
-            json.dumps(old_model_data, cls=DjangoJSONEncoder),
-            ignorenonexistent=True,
-        )
-        new_deserialized_objects = serializers.deserialize(
-            'json',
-            json.dumps(new_model_data, cls=DjangoJSONEncoder),
-            ignorenonexistent=True,
-        )
-
-        old = next(old_deserialized_objects, None)
-        if old is not None:
-            old = old.object
-        new = next(new_deserialized_objects, None)
-        if new is not None:
-            new = new.object
+        old = cls._deserialize_from_state(payload_dict, state='old')
+        new = cls._deserialize_from_state(payload_dict, state='new')
         return {'old': old, 'new': new}
 
     @classmethod
-    def _build_model_serializer_data(cls, payload: Dict, state: str):
-        """Reformat serialized data into shape as expected
-        by the Django model deserializer.
-        """
+    def _is_up_to_date(cls, app: str, db_version_id: int) -> bool:
+        """Check if the db version id from django migrations is the latest for the given app"""
+        newer_migration_exists = MigrationRecorder.Migration.objects.filter(
+            app=app, id__gt=db_version_id
+        ).exists()
+        return not newer_migration_exists
```

Comment:
This code assumes that ids for the newer migrations are larger then for the older migrations. Squashed migrations get a new (means larger) id. If the serialized form was created for the migration before the squashed migration is applied it will be considered as an outdated form. If the squash was done for some old migrations (that is for the migrations range that does not involve the latest migration for the application) than this behaviour is correct as the serialized form is for the outdated version of the entity. If the squash is for the latest (unsquashed) migration then this behaviour is also ok as the serialized form is the same as for the latest version of the entity.

Regarding the reversal I think an improvement should be done here. When we save the serialized form we should store not the global DB version but the latest migration id of the corresponding application. Then here we should check not if newer migrations exist but if the latest application migration is equal to the one is stored. This way we can detect both forward and backward migrations. How do you think?